lib/rust/flexer/src/generate.rs

```diff
@@ -0,0 +1,531 @@
+//! This file contains utilities for generating rust code from lexer definitions, allowing the
+//! flexer to be specialised for a specific language.
+
+use crate::prelude::*;
+
+use crate::automata::dfa::DFA;
+use crate::automata::dfa::RuleExecutable;
+use crate::automata::state::Identifier;
+use crate::automata::state::State;
+use crate::group::Group;
+use crate::group;
+
+use enso_macro_utils::repr;
+use proc_macro2::Literal;
+use quote::*;
+use std::hash::BuildHasher;
+use std::result::Result;
+use syn::*;
+use std::fmt;
+
+use crate as flexer;
+use wasm_bindgen::__rt::core::fmt::Formatter;
+
+
+// =======================
+// === Code Generation ===
+// =======================
+
+/// Generate specialized code for the provided lexer `definition`.
+///
+/// This specialized code is a highly-optimised and tailored lexer that dispatches based on simple
+/// code-point switches, with no dynamic lookup. This means that it is very fast, and very low
+/// overhead.
+pub fn specialize
+( definition:&impl flexer::State
+, state_type_name:impl Into<String>
+, output_type_name:impl Into<String>
```

Comment:
When using one paremter per line form, they should be aligned by `:` (and then `:` gets its spaces around).
This also applies to other occurrences.